relay-sampling/src/lib.rs

```diff
@@ -115,6 +116,51 @@ impl EqCondition {
     }
 }
 
+macro_rules! impl_cmp_condition {
+    ($struct_name:ident, $operator:tt) => {
+        #[derive(Debug, Clone, Serialize, Deserialize)]
+        pub struct $struct_name {
+            pub name: String,
+            pub value: Number,
+        }
+
+        impl $struct_name {
+            fn matches_event(&self, event: &Event) -> bool {
+                self.matches(event)
+            }
+            fn matches_trace(&self, trace: &TraceContext) -> bool {
+                self.matches(trace)
+            }
+
+            fn matches<T: FieldValueProvider>(&self, value_provider: &T) -> bool {
+                let value = match value_provider.get_value(self.name.as_str()) {
+                    Value::Number(x) => x,
+                    _ => return false
+                };
+
+                // Try various conversion functions in order of expensiveness and likelihood
+                // - as_i64 is not really fast, but most values in sampling rules can be i64, so we could
+                //   return early
+                // - f64 is more likely to succeed than u64, but we might lose precision
+                if let (Some(a), Some(b)) = (value.as_i64(), self.value.as_i64()) {
```

Comment:
I _think_ this should be fine but tbh i'm not super confident. I'm trying to think of cases where both values don't have decimal points, but can't be cast to the same int type:

e.g.: `self.value` is negative, event `value` is u64::MAX (`self.value` can't be cast to u64, and event `value` can't be cast to i64)

but i think in those cases we should be fine with casting those values to floats for comparison...